compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -57,6 +57,7 @@ pub(crate) fn codegen_fn<'tcx>(
 
         match &mir.basic_blocks[START_BLOCK].terminator().kind {
             TerminatorKind::InlineAsm {
+                asm_macro: _,
```

Comment:
Maybe make this check for NakedAsm, such that anything else will cause the unreachable!() below to be hit?